request-manager/graph/graph.go

```diff
@@ -44,14 +45,14 @@ type Node struct {
 	SequenceRetryWait string                 // The time to sleep between sequence retries
 }
 
-// Asserts that g is a valid graph (according to Grapher's use case).
-// Ensures that g is acyclic, is connected (not fully connected),
-// and edge map matches reverse edge map.
+// IsValidGraph asserts that g is a valid graph by ensuring that
+// g is acyclic and connected (not fully connected), and that the edge
+// map matches the reverse edge map.
 func (g *Graph) IsValidGraph() bool {
 	return !g.hasCycles() && g.isConnected() && g.edgesMatchesRevEdges()
 }
 
-// Get out edges of node (node id --> Node)
+// GetNext returns the out edges of a node (node id --> Node)
 func (g *Graph) GetNext(n *Node) map[string]*Node {
```

Comment:
No callers use the key, only the value. So the return can probably be `[]*Node`.

Reply:
Is it really okay for `GetNext` and `GetPrev` to have different return types like that?

Reply:
No, that'd not be ideal, but I think (iirc the code) they can both return `[]*Node`?